solve.py

```diff
@@ -103,5 +103,19 @@ def add_general_arguments_and_parse_settings(default_inst_file: str = '0010.txt'
         logger.info("")
         alg.method_statistics()
         alg.main_results()
+    elif settings.alg == "vnd":
+        # Which neighborhoods do we want to use here?
+        alg = GVNS(solution, [], random.sample([
+            Method("li_2opt_best", CBTSPSolution.local_improve, (Neighbor.KOPT2, Step.BEST)),
+            Method("li_3opt_best", CBTSPSolution.local_improve, (Neighbor.KOPT3, Step.BEST)),
+            Method("li_xchg_best", CBTSPSolution.local_improve, (Neighbor.XCHG, Step.BEST)),
+            Method("li_smove_best", CBTSPSolution.local_improve, (Neighbor.SMOVE, Step.BEST)),
+            Method("li_sblock_best", CBTSPSolution.local_improve, (Neighbor.SBLOCK, Step.BEST)),
+            Method("li_2.5opt_best", CBTSPSolution.local_improve, (Neighbor.KOPT2HALF, Step.BEST))
+        ], 6), [], ownsettings)
+        alg.vnd(solution.copy()) #Needs to be a copy or update_incumbent in algorithm will override the argument
```

Comment:
Das führt ja nur eine Iteration von VND durch und keine vollständige Suche, oder? Denke richtig wäre Aufruf von alg.run() mit leerer shake Funktion. Aber werde heute Nacht local search noch fertig machen, da kann ich das gleich dazu nochmal anschauen.

Reply:
Ein Aufruf von `alg.vnd(...)` entspricht genau dem VND-Algorithmus in den Folien (+ timeout handling). `alg.run()` mit leerem shaking macht glaub ich nicht wirklich was anderes, weil gleich am Anfang von `gvns(...)` steht:
```
sol2 = sol.copy()
if self.vnd(sol2) or not self.meths_sh:
    return
```
d.h. ohne shaking macht es auch nur einmal VND. Einziger Unterschied ist, dass man über `run` auch gleich noch construction heuristics anwenden lassen kann, dafür wärs viellieicht sinnvoll.

Reply:
Habs umgestellt, jetzt wird über `alg.run()` rconst+VND ausgeführt.